interview-questions.md

```diff
@@ -7,62 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: The && (and) operator and the || (or) operator. With the && operator, both statements have to be true in order to return the Boolean value of true. Forthe || operator, only one statement has to evaluate to true to return the value of true, and if the first statement is true, it may not run the second statement.
 
-  Researched answer:
+  Researched answer: The || operator can be used with operands that aren't Boolean, but the return will be converted to a Boolean primitive. The && operator has higher precedence than the || operator, which means it will execute first.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is the location of an element in an array and has a 0-base. For example, myArray = ["apple", "banana", "strawberry"]. Here, index[0] = apple. index[1] = "banana", and index[2] = "strawberry".  A value is what a something is equal too such as a variable. For example, temp1 = 75, 75 would be the value of temp1 as there is no location and no index of temp1.
 
-  Researched answer:
+  Researched answer: Strings also have an index for each letter and space to account for the placement. 0-based indexing means you start at 0. Values are used to define what is being held in the element/ index.
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is the process of repeating a process over and over again. In coding, we can write functions and iterate that function through each element of an array.
 
-  Researched answer:
+  Researched answer: Iteration is great when you want to execute a function multiple times with differenct inputs. An example of this is using a For loop with arrays.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Proper pair programming involves having a driver and navigator. The driver is the hands on the keyboard person, who is actually typing out code. The navigator is the person who is looking at the tasks, brainstorming ideas with the driver about how to approach the tasks, and gently calling out coding errors. 
 
-  Researched answer:
+  Researched answer: It is two people working collaborately on code to produce a working program. This allows for the workload to be split up and higher communication levels within a team, creating a cohesive environment for productivity and enjoyment. After a set amount of time or tasks, partners want to switch roles of who is navigating and who is driving, which allows for growth as well. The ability to verbalize how you approach a problem and synthezise your partners viewpoint with your own will only help you to grow and develop more skills for this profession.
```

Comment:
Awesome work in this section!  Answers are worded wonderfully and well thought out!